src/solver/variable/economy/nearPriceCap.h

```diff
@@ -0,0 +1,243 @@
+            pValuesForTheCurrentYear[numSpace].reset();
+
+        NextType::simulationBegin();
+    }
+
+    void yearBegin(unsigned int year, unsigned int numSpace)
+    {
+        // Reset the values for the current year
+        pValuesForTheCurrentYear[numSpace].reset();
+
+        NextType::yearBegin(year, numSpace);
+    }
+
+    void yearEndBuild(State& state, unsigned int year)
+    {
+        NextType::yearEndBuild(state, year);
+    }
+
+    void yearEnd(unsigned int year, unsigned int numSpace)
+    {
+        // Time related aggregations to daily, weekly, monthly,and annual results
+        // are computed by summing hourly results
+        pValuesForTheCurrentYear[numSpace].computeStatisticsForTheCurrentYear();
+
+        NextType::yearEnd(year, numSpace);
+    }
+
+    void computeSummary(std::map<unsigned int, unsigned int>& numSpaceToYear,
+                        unsigned int nbYearsForCurrentSummary)
+    {
+        for (unsigned int numSpace = 0; numSpace < nbYearsForCurrentSummary; ++numSpace)
+        {
+            // Merge all those values with the global results
+            AncestorType::pResults.merge(numSpaceToYear[numSpace] /*year*/,
+                                         pValuesForTheCurrentYear[numSpace]);
+        }
+
+        NextType::computeSummary(numSpaceToYear, nbYearsForCurrentSummary);
+    }
+
+    void hourForEachArea(State& state, unsigned int numSpace)
+    {
+        double mrgPrice = -state.hourlyResults->CoutsMarginauxHoraires[state.hourInTheWeek];
+
+        if (mrgPrice > unsuppliedEnergyCost - margin + eps)
+            pValuesForTheCurrentYear[numSpace][state.hourInTheYear] = 1.;
+
+        NextType::hourForEachArea(state, numSpace);
+    }
+
+    Antares::Memory::Stored<double>::ConstReturnType retrieveRawHourlyValuesForCurrentYear(
+      unsigned int,
+      unsigned int numSpace) const
+    {
+        return pValuesForTheCurrentYear[numSpace].hour;
+    }
+
+    void localBuildAnnualSurveyReport(SurveyResults& results,
+                                      int fileLevel,
+                                      int precision,
+                                      unsigned int numSpace) const
+    {
+        // Initializing external pointer on current variable non applicable status
+        results.isCurrentVarNA = AncestorType::isNonApplicable;
+
+        if (AncestorType::isPrinted[0])
+        {
+            // Write the data for the current year
+            results.variableCaption = VCardType::Caption();
+            results.variableUnit = VCardType::Unit();
+            pValuesForTheCurrentYear[numSpace].template buildAnnualSurveyReport<VCardType>(
+              results, fileLevel, precision);
+        }
+    }
+
+private:
+    //! Intermediate values for each year
+    typename VCardType::IntermediateValuesType pValuesForTheCurrentYear;
+    unsigned int pNbYearsParallel;
+    const double eps = 1e-2;
+    const double margin = 5.;
```

Comment:
- `constexpr`
- at the top of the file, outside the class definition (no need for 1 `eps` per class instance)